lib/howl/commands/app_commands.moon

```diff
@@ -302,36 +313,54 @@ command.register
 -----------------------------------------------------------------------
 
 launch_cmd = (working_directory, cmd) ->
-  shell = howl.sys.env.SHELL or '/bin/sh'
-  p = Process {
-    :cmd,
-    :shell,
-    read_stdout: true,
-    read_stderr: true,
-    working_directory: working_directory,
-  }
-
-  buffer = ProcessBuffer p
-  editor = app\add_buffer buffer
-  editor.cursor\eof!
-  buffer\pump!
+  dispatch.launch ->
```

Comment:
Another way (coming full circle further) might be to provide two separate functions in the command definition - the interactive and non-interactive parts. This would subsume the `interactive` boolean. The non interactive part would run in its own coroutine. This is not that different from my earlier suggesting of returning a function from the handler (because you split your code into two functions either way), but it is better in other ways because two distinct keys in the command definition make the meaning clearer I think.

To develop this idea further:

```moonscript
command.register:
   name: 'command'
   interactor: -- (optional) either a function or a factory table, blocks the command line while running
   handler: -> -- function that runs in a separate coroutine
```

The output of interactor would be fed into handler (details tbd).